src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetProcessName.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, EntryPoint = "QueryFullProcessImageNameW", ExactSpelling = true)]
+        private static extern bool QueryFullProcessImageName(
+            IntPtr hProcess,
+            uint dwFlags,
+            ref char lpBuffer,
+            ref uint lpdwSize);
+
+        private const int MAX_PROCESSNAME_LENGTH = 1024;
```

Comment:
where does this value come from? Could you please add a comment with a link to a doc?

Reply:
Yes, good idea. I searched around the repo and grabbed it from somewhere.  I'll search around again and make a note in the code!

Reply:
@adamsitnik - I can't quite remember where I found this, but on reflection, it doesn't look adequate. `MAX_PATH` in `minwindef.h`  is set to `260`, but of course, it's possible to enable long path support, but I don't know if long path support affects actual allowable filename lengths.

It would seem odd to allocate megabytes (especially on the stack) just to allow for the longest name ever imaginable, but the alternative also seems odd and unnecessary; start with a small buffer and continually increase it if the call to `QueryFullProcessImageName` writes to `ref length` with a value of our max buffer size.

Reply:
We should handle `ERROR_INSUFFICIENT_BUFFER` in similar way we do it for other path-related APIs. Since handling it is a PITA I am going to push a fix to your branch, I hope you don't mind.